src/modules/referral/services/cron-service.ts

```diff
@@ -3,43 +3,64 @@ import { Repository } from "typeorm";
 import { InjectRepository } from "@nestjs/typeorm";
 import { AppConfig } from "../../configuration/configuration.service";
 import { Deposit } from "../../scraper/model/deposit.entity";
-import { EnhancedCron } from "../../../utils";
-import { updateStickyReferralAddresses } from "../services/queries";
+import { EnhancedCron, wait } from "../../../utils";
+import { updateStickyReferralAddressesQuery } from "../services/queries";
+import { StickyReferralAddressesMechanism } from "src/modules/configuration";
 
 @Injectable()
 export class ReferralCronService {
   private logger = new Logger(ReferralCronService.name);
+  private semaphore = false;
```

Comment:
This is unlikely to happen, but in case the previous execution of the cron doesn't finish, don't trigger a new one.

Reply:
Just out of curiosity, what bad things would this cause if two ran at the same time?

Just thinking about the other extreme, where one run hangs because some promise never returned, and then the cron just silently stops. I don't think this is _likely_ but it seems possible at least.

Reply:
> Just out of curiosity, what bad things would this cause if two ran at the same time?

Firstly, the tables impacted by these 2 intensive queries will become very slow and most probably they won't finish until future execution are triggered. This will put pressure on the DB. In time the resources usage will start to increase significantly (cpu and memory up to 100%)